read-fonts/src/tables/glyf.rs

```diff
@@ -741,15 +753,21 @@ impl<'a> ContourIter<'a> {
             // Establish our first move and the starting index for our waltz over the points
             first_move = contour.points[0];
             if contour.flags[0].is_off_curve_quad() {
-                let maybe_oncurve_ix = contour.points.len() - 1;
+                let maybe_oncurve_ix = match contour.offcurve_first_mode {
+                    OffCurveFirstMode::Back => contour.points.len() - 1,
+                    OffCurveFirstMode::Front => (start_ix + 1).min(contour.points.len() - 1),
```

Comment:
why do you only look at the immediately next or previous point and not also the following ones? TrueType curves with implied oncurve points may contain several offcurves preceding an oncurve. The fonttools PointToSegmentPen looks forward until it finds any oncurve point and uses that as the moveTo. If it can't find any oncurves, it takes the midpoint between the first and last off-curves.

Reply:
Good catch!

I think I should rename the variants to more clearly indicate their goals which are really to match the behavior of existing systems. The back mode matches freetype. Assuming the original source was correct freetype doesn't do the walk until oncurve bit. Front matches hb-draw, if hb-draw does so then I'm missing a test and should add + fix.

Reply:
For the point stream:

```xml
<pt x="278" y="710" on="0"/>
<pt x="278" y="470" on="0"/>
<pt x="998" y="470" on="1"/>
<pt x="998" y="710" on="0"/>
<pt x="750" y="500" on="0"/>
```

Our internal hb-draw based tool produces:

```
M278,-590
Q278,-470 998,-470
Q998,-710 874,-605
Q750,-500 514,-605
Q278,-710 278,-590
Z
```

Negative is because it just flips y. The starting move to y of 590 seems to support the midpoint between first and second rather than scanning forward multiple steps to find the oncurve at 998,470.

Reply:
Added a test based on the 2-off, 1-on, 2-off case above.

Reply:
ok thanks for confirming that.